src/DataStructures/DataBox/DataBoxTag.hpp

```diff
@@ -223,17 +226,32 @@ struct item_type_impl {
                 "cannot be modified.  You probably wanted const_item_type.");
   using type = DataBox_detail::storage_type<Tag, TagList>;
 };
+
+template <typename Tag, typename TagList, typename = std::nullptr_t>
+struct const_item_type_impl {
+  using type = std::decay_t<decltype(DataBox_detail::convert_to_const_type(
+      std::declval<DataBox_detail::storage_type<Tag, TagList>>()))>;
+};
+
+CREATE_HAS_TYPE_ALIAS(type)
+CREATE_HAS_TYPE_ALIAS_V(type)
+
+template <typename Tag>
+struct const_item_type_impl<Tag, NoSuchType,
+                            Requires<is_base_tag_v<Tag> and has_type_v<Tag>>> {
+  using type = tmpl::type_from<Tag>;
+};
```

Comment:
This is the bug. You are adding undefined behavior into base tags, and making base tags behave differently if passed into `const_item_type` and `item_type` in a way that makes no sense from a user perspective. It also means that now there is a semi-implicit relation between base tags and simple tags, and their type, that is not a simple relationship. You are *literally breaking all the guarantees of base tags*